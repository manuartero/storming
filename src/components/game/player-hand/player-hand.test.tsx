import { render } from "@testing-library/react";
import PlayerHand from "./player-hand";

describe("<PlayerHand />", () => {
  test("render: match snapshot", () => {
    const { asFragment } = render(
      <PlayerHand cards={playerHand} onClick={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

const playerHand: PlayerHand = [
  {
    card: {
      cardType: "actionCard",
      action: "move",
      owner: "player",
      cardId: "player_move_A",
    },
    status: "available",
  },
  {
    card: {
      cardType: "actionCard",
      action: "move",
      owner: "player",
      cardId: "player_move_B",
    },
    status: "available",
  },
  {
    card: {
      cardType: "actionCard",
      action: "build",
      owner: "player",
      cardId: "player_build_A",
    },
    status: "available",
  },
];
