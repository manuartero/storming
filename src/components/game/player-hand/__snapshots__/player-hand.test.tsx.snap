// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerHand /> render: match snapshot 1`] = `
<DocumentFragment>
  <div
    class="player-hand"
  />
</DocumentFragment>
`;
